src/analyzing/AST_to_IL.ml

```diff
@@ -1804,24 +1949,27 @@ and stmt_aux env st =
   (* Java: synchronized (E) S *)
   | G.OtherStmtWithStmt (G.OSWS_Block _, [ G.E objorig ], stmt1) ->
       (* TODO: Restrict this to a syncrhonized block ? *)
-      let ss, _TODO_obj = expr_with_pre_stmts env objorig in
-      ss @ stmt env stmt1
+      let env, ss, _TODO_obj = expr_with_pre_stmts env objorig in
+      let env, new_stmts = stmt env stmt1 in
+      (env, ss @ new_stmts)
   (* Rust: unsafe block *)
   | G.OtherStmtWithStmt (G.OSWS_Block ("Unsafe", tok), [], stmt1) ->
       let todo_stmt = fixme_stmt ToDo (G.TodoK ("unsafe_block", tok)) in
-      todo_stmt @ stmt env stmt1
+      let env, new_stmts = stmt env stmt1 in
+      (env, todo_stmt @ new_stmts)
   | G.OtherStmt (OS_Async, [ G.S stmt1 ]) ->
       let todo_stmt = fixme_stmt ToDo (G.TodoK ("async", G.fake "async")) in
-      todo_stmt @ stmt env stmt1
+      let env, new_stmts = stmt env stmt1 in
+      (env, todo_stmt @ new_stmts)
   | G.OtherStmt _
   | G.OtherStmtWithStmt _ ->
-      todo (G.S st)
-  | G.RawStmt _ -> todo (G.S st)
+      todo env.stmts (G.S st)
+  | G.RawStmt _ -> todo env.stmts (G.S st)
 
 and for_each env tok (pat, tok2, e) st =
   let cont_label_s, break_label_s, st_env = mk_break_continue_labels env tok in
-  let ss, e' = expr_with_pre_stmts env e in
-  let st = stmt st_env st in
+  let env, ss, e' = expr_with_pre_stmts env e in
+  let _, st = stmt st_env st in
```

Comment:
Originally `env` and `st_env` shared the `stmts` reference, could that affect things?

Reply:
Given that we throw away the returned env here.